json-streaming/src/main/scala/org/apache/pekko/stream/connectors/json/impl/JsonStreamReader.scala

```diff
@@ -69,8 +69,8 @@ private[pekko] final class JsonStreamReader(path: JsonPath) extends GraphStage[F
         }
 
         if (buffer.nonEmpty) {
-          emitMultiple(out, buffer)
-          buffer = Queue.empty[ByteString]
+          emitMultiple(out, buffer.toSeq.iterator)
```

Comment:
So I just noticed this now and I am thinking that the `List(ByteString(value.toString))` variant is better because while there may be a trivial box for a single `List` (which might get inlined anyways), the `buffer.toSeq.iterator` is basically going to copy the entire collection which is much worse especially considering that the size of this collection can be quite large

Reply:
@pjfanning You can revert this line now since buffer is immutable